test/scenarios/overrride_driver/molecule/default/playbook.yml

```diff
@@ -5,3 +5,11 @@
     - name: Dummy task
       command: /bin/true
       changed_when: false
+    - name: check presence of from-custom-dockerfile
```

Comment:
This looks like a good strategy then! However, the `override_driver` is now testing two things: custom dockerfile and overriding the driver. Can we separate these tests out into separate scenarios? Let's ignore the fact that our functional tests take 10000000 years to run and assume we'll get that improved soon :) You'll also need to add a a test in https://github.com/ansible/molecule/blob/master/test/functional/docker/test_scenarios.py to drive it as well.

Reply:
@decentral1se I am really concerned about the time needed to run the test and if we do not combine them we will be doomed not to be able to run the tests.

I am more pragmatic here, i do care to test multiple execution paths and provide a good code (branch) coverage. I do not care if a specific test is testing more than one aspect, if it fails it should be easy to stop what went wrong.

On the other hand "override" seems like a good candidate, we are also overriding the default template ;)

Reply:
@decentral1se I tried to follow your recomandation and add a new scenarion and this broke the doors of hell... take a look at how the tests are written on https://github.com/ansible/molecule/blob/0627ca213048a5bd5ea671f6b14eee2a7624f867/test/functional/test_command.py#L394

I have to modify half of them to allow a new scenario to be added because they check for the list of expected scenarios.

I have the impression that that stuff with test isolation makes no sense, we cannot really add new scenarios without affecting almost everythign else.

Isolation is good for unittesting but for functional, i doubt is doable.